File: agent/bindings/kotlin/lib/src/main/kotlin/com/sourcegraph/cody/agent/protocol_generated/ProtocolCommand.kt
@file:Suppress("FunctionName", "ClassName", "unused", "EnumEntryName", "UnusedImport")
package com.sourcegraph.cody.agent.protocol_generated;

data class ProtocolCommand(
  val title: TitleParams,
  val command: String,
  val tooltip: String? = null,
  val arguments: List<Any>? = null,
)

File: agent/bindings/csharp/ProtocolCommand.cs
using System.Text.Json.Serialization;

namespace Cody.Core.Agent.Protocol
{
  public class ProtocolCommand
  {
    [JsonProperty(PropertyName = "title")]
    public TitleParams Title { get; set; }
    [JsonProperty(PropertyName = "command")]
    public string Command { get; set; }
    [JsonProperty(PropertyName = "tooltip")]
    public string Tooltip { get; set; }
    [JsonProperty(PropertyName = "arguments")]
    public Object[] Arguments { get; set; }
  }
}
